### ISSUE TITLE
Release Regularly

### ISSUE THREAD

**Issue by jsmartt** [HewlettPackard/oneview-sdk-ruby#156]
We really need to start releasing more regularly, instead of major releases after we've implemented 20-30 features. There are a lot of bug fixes and enhancements that are being held off because we're waiting to release a big version.

That said, we do want to limit the number of major releases we do, and chunk together as many breaking changes as we can into a single major release. I think we can do this much more effectively if we use a separate branch to hold breaking changes, and only merge breaking changes into that branch until we're ready to release, at which point we would merge the breaking_changes branch into master.  This way we can continue to merge small changes into master and release regularly, but control the release of breaking changes.

Thoughts?

**Comment by jsmartt** [comment]
BTW, I think it would be good to get any open PRs we need merged in (before adding more to the list) and doing a release as soon as possible. If we need to back out the breaking changes in #93 we can, but we need to get the other updates released.

**Comment by jsmartt** [comment]
@aalexmonteiro , @fgbulsoni , @ricardogpsf , @tmiotto 

Curious what other people's thoughts are on this?
- What do we need to merge (if anything) before making a release? (I think there are a few open PRs that are ready)
- Are we confident with the integration tests that have run, or does someone need to run a full set before the release?
- When can we expect to be ready for the release?

**Comment by aalexmonteiro** [comment]
- What do we need to merge (if anything) before making a release? (I think there are a few open PRs that are ready)
If all open PRs are approved. The features of the i3s will be partially released. Today I'm going to open another PR (Development Plan) and  believe that soon we can release all features of the i3s.

- Are we confident with the integration tests that have run, or does someone need to run a full set before the release?

The test integrations for i3s, three endpoints could not be tested on the current version of i3s running on the actual hw in India due to some bugs in the current version. But the other tests are ok.


- When can we expect to be ready for the release?

If you think it's better to release a new version soon. This version would have the i3s partially made available. But I do not see problems in being partially available.

**Comment by ricardogpsf** [comment]
I like idea of the start releasing more regularly.

The PR #148  that was opened by me is ready for merge (unless someone suggests improvements) and the tests are running good.

**Comment by ricardogpsf** [comment]
Will the breaking changes (from the separeted branch) be scheduled for merge with master?
And will the new big features/changes be merged in any moment in minor versions? Or will be scheduled too?

We could define the changes that are viable for merge in each version (if in scheduled major version or in minor version).

**Comment by jsmartt** [comment]
@ricardogpsf , I think we can definitely schedule when to merge in the breaking changes, but I think we should only do that step if we're planning on releasing it pretty much immediately afterwards. The main reason for having that branch separated is for cases like we're in now, where we have  bunch of bug fixes and feature additions that are being held up by a small breaking change that we don't necessarily want to have to release yet. I'm not proposing separate branches for feature adds and bug-fixes; just majors.

New features and bug fixes can be released on a schedule, but I think it just makes sense to release those as soon as possible. We can certainly have a goal of once a week (for instance) if we want to do these on a schedule, but we don't need to stick to that exact timing. But bug fixes need to be release ASAP, and why not release new features ASAP too?

**Comment by ricardogpsf** [comment]
@jsmartt Sounds good. The breaking changes will be merged for a specific branch and will be merged with master when releasing the major version. I liked this. =)
